arduino-core/src/processing/app/debug/Compiler.java

```diff
@@ -1042,6 +1051,37 @@ void runRecipe(String recipe) throws RunnerException, PreferencesMapException {
     }
     execAsynchronously(cmdArray);
   }
+  
+  //7. Save the .hex file
+  void saveHex() throws RunnerException {
+    PreferencesMap dict = new PreferencesMap(prefs);
+    dict.put("ide_version", "" + BaseNoGui.REVISION);
+
+    String[] cmdArray;
+    try {
+      String tmp_file = prefs.getOrExcept("recipe.output.tmp_file");
+      tmp_file = StringReplacer.replaceFromMapping(tmp_file, dict);
+      String save_file = prefs.getOrExcept("recipe.output.save_file");
+      save_file = StringReplacer.replaceFromMapping(save_file, dict);
+
+      File hexFile = new File(prefs.get("build.path") + "/" + tmp_file);
+      File saveFile = new File(sketch.getFolder().getAbsolutePath() + "/" + save_file);
+
+      FileReader in = new FileReader(hexFile);
+      FileWriter out = new FileWriter(saveFile);
+
+      int c;
+      while ((c = in.read()) != -1)
+        out.write(c);
+
+      in.close();
+      out.close();
+      
```

Comment:
Also for AVR we would like to save the .eep ??

=======================================
  Arnav Gupta (championswimmer)
  LinkedIn <http://in.linkedin.com/in/arnavgupta/> | Web
<http://championswimmer.in> | Blog <http://blog.championswimmer.in> |
Twitter <http://twitter.com/championswimmer> | Facebook
<http://facebook.com/championswimmer> | Github
<http://github.com/championswimmer>


On 18 February 2015 at 16:08, Cristian Maglie <notifications@github.com>
wrote:

> In arduino-core/src/processing/app/debug/Compiler.java
> <https://github.com/arduino/Arduino/pull/2567#discussion_r24892263>:
>
> > +      String save_file = prefs.getOrExcept("recipe.output.save_file");
> > +      save_file = StringReplacer.replaceFromMapping(save_file, dict);
> > +
> > +      File hexFile = new File(prefs.get("build.path") + "/" + tmp_file);
> > +      File saveFile = new File(sketch.getFolder().getAbsolutePath() + "/" + save_file);
> > +
> > +      FileReader in = new FileReader(hexFile);
> > +      FileWriter out = new FileWriter(saveFile);
> > +
> > +      int c;
> > +      while ((c = in.read()) != -1)
> > +        out.write(c);
> > +
> > +      in.close();
> > +      out.close();
> > +
>
> May you add a message for verbose compile? Something like:
>
> Copying output file xxxxxx.hex to /xxx/yyy/zzz
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/arduino/Arduino/pull/2567/files#r24892263>.
>